packages/sdk/src/purchase/delegation.ts

```diff
@@ -0,0 +1,69 @@
+import { ethers, Signer, providers } from "ethers";
+
+import { IDelegationRegistry } from "../contracts/IDelegationRegistry";
+import { IDelegationRegistry__factory } from "../contracts/factories/IDelegationRegistry__factory";
+
+/**
+ * Define the registry for delegating a hot wallet to mint/claim airdrops on behalf of a vault wallet
+ * @see https://delegate.cash/
+ */
+export const DELEGATION_REGISTRY = "0x00000000000076a84fef008cdabe6409d2fe638b";
+
+/**
+ * @summary Error thrown when a user is a delegate on behalf of multiple vaults.
+ * @description Currently, only a single vault can be delegated to at a time.
+ * This error is thrown when a user has delegated to multiple vaults, and provides
+ * an easy way for SDK consumers to catch this error and display a helpful message
+ * to the user.
+ */
+export class MultipleDelegationsError extends Error {
```

Comment:
My inclination would be to update `getDelegateVault` to be `getDelegateVaults` and allow the SDK consumer to decide how they want to handle the case where there's multiple vaults. We could set up purchase so that it takes an optional vault that's provided by the caller and then we don't need to care about multiple vaults in the SDK itself.

For context, I think we just didn't want to figure out the UX for users with multiple vaults but since we're not dealing with the UX here directly I think it's fine to leave this more flexible